### ISSUE TITLE
Radarr V3: Issue with "Status Column"

### ISSUE THREAD

**Issue by dramirvf** [Radarr/Radarr#5252]
There is a problem with **"Status Column"** as it is not sortable. If **"Status"** is clicked to sort, nothing actually happens.
Being sorted according to status is very essential here, as there are hundreds of missing items, which some of them are currently being downloaded. I want to only perform actions on missing items which are not being downloaded at the moment. If the column is not sorted according to status, there is no way to achieve this.

Since sorting column doesn't work, I tried to exclude **"Downloading"** items through status filter. Unfortunately, **"Status Filter"** has other values, which actually demonstrate the **"Release Status"**, not the status of the item in Radarr database.

I have shown the issue in the following image:
![image](https://user-images.githubusercontent.com/40552131/96875784-81273100-1484-11eb-9d97-285e4b2ea01f.png)

**Comment by dako98** [comment]
Sorting by Status works fine for me. 
The custom filter does indeed show Release Status values, rather than Radarr status.

Also on Radarr Version: 3.0.0.3966

**Comment by dramirvf** [comment]
> Sorting by Status works fine for me.
> The custom filter does indeed show Release Status values, rather than Radarr status.
> 
> Also on Radarr Version: 3.0.0.3966

When the movie is downloaded, **"Sorting by Status"** works fine. When the movie is missing, **"Sorting by Status"** does not work as expected. In fact, it does not sort the following value properly: **"Missing", "Downloading", "Error", "Paused"**, etc.

To reproduce, just change your filter to missing and make sure some of missing videos are being downloaded in download client.

**Comment by dramirvf** [comment]
> Sorting by Status works fine for me.
> The custom filter does indeed show Release Status values, rather than Radarr status.
> 
> Also on Radarr Version: 3.0.0.3966

I know that the custom filter does indeed show **"Release Status"** values, but it is much better to have another custom filter to show the current availability status or as you say **"Radarr Status"**. Additionally, using the word "**Status"** is confusing for the user. Because there are two distinct applications of this word in the current UI; so it is better to use unique words for each across different sections of UI.

**Comment by geogolem** [comment]
The confusion regarding custom filters of status and release status is fixed in a PR that is yet to be merged: https://github.com/Radarr/Radarr/pull/4924

Since that issue is fixed - lets keep this issue focused on the problems sorting the "Status" column with values, "Missing", "Downloading", "Error", "Paused" --> I have not tried to reproduce it yet but it sounds like a legitimate issue.

**Comment by geogolem** [comment]
@dramirvf : lets bring that discussion here so that we can better keep track of it..

I am just pasting @austinwbest  comment here because i think it does clarify things. 

> So the status changes from missing to downloading as we are not adding another column for that temporary change and it is directly related to missing. It isn't missing if it is downloading. For things being downloaded there is the activity page. For things that are downloaded being switched to quality, it is by many user request and is that way as the quality is more informative than everything saying downloaded.

>It is a given that it is downloaded or available of there is a quality.

**Comment by dramirvf** [comment]
@geogolem I just put a screenshot of our latest conversations here. We'll continue the discussion here.

![radarr_1](https://user-images.githubusercontent.com/40552131/99138028-fbda0b00-2642-11eb-910d-e80f338b6ba4.jpg)

![radarr_2](https://user-images.githubusercontent.com/40552131/99138032-04324600-2643-11eb-936e-d3b87198e4af.jpg)

![radarr_3](https://user-images.githubusercontent.com/40552131/99138040-0b595400-2643-11eb-8a7f-6a8832c216c7.jpg)

**Comment by dramirvf** [comment]
> @dramirvf : lets bring that discussion here so that we can better keep track of it..
> 
> I am just pasting @austinwbest comment here because i think it does clarify things.
> 
> > So the status changes from missing to downloading as we are not adding another column for that temporary change and it is directly related to missing. It isn't missing if it is downloading. For things being downloaded there is the activity page. For things that are downloaded being switched to quality, it is by many user request and is that way as the quality is more informative than everything saying downloaded.
> 
> > It is a given that it is downloaded or available of there is a quality.
> 
> This makes sense. I need to test it out, but if that is the case, then this issue is really a non issue - with the exception of Status column sorting....
> 
> regarding AvailabilityStatus in custom filters -that is being handled by another issue: #2074 which has a PR and is almost complete but still a WIP - anythnig regarding availablitystatus custom filters should be discussed at #2074
> 
> Lets keep this issue focused strictly on Status Column Sorting (but feel free to further discuss the Missing+Downloading issue here if you still think it is an issue but based on @austinwbest comment I think it is a non issue)

![image](https://user-images.githubusercontent.com/40552131/99138458-d8fd2600-2645-11eb-948c-68effc08d2cf.png)

I couldn't find a custom filter to show whether a movie itself is either **"Downloaded"** or **"Not Downloaded"**.
Do you mean the following filter?

![image](https://user-images.githubusercontent.com/40552131/99138552-94be5580-2646-11eb-904f-ca60ffea8de4.png)

**Comment by geogolem** [comment]
> > @dramirvf : lets bring that discussion here so that we can better keep track of it..
> > I am just pasting @austinwbest comment here because i think it does clarify things.
> > > So the status changes from missing to downloading as we are not adding another column for that temporary change and it is directly related to missing. It isn't missing if it is downloading. For things being downloaded there is the activity page. For things that are downloaded being switched to quality, it is by many user request and is that way as the quality is more informative than everything saying downloaded.
> > 
> > 
> > > It is a given that it is downloaded or available of there is a quality.
> > 
> > 
> > This makes sense. I need to test it out, but if that is the case, then this issue is really a non issue - with the exception of Status column sorting....
> > regarding AvailabilityStatus in custom filters -that is being handled by another issue: #2074 which has a PR and is almost complete but still a WIP - anythnig regarding availablitystatus custom filters should be discussed at #2074
> > Lets keep this issue focused strictly on Status Column Sorting (but feel free to further discuss the Missing+Downloading issue here if you still think it is an issue but based on @austinwbest comment I think it is a non issue)
> 
> ![image](https://user-images.githubusercontent.com/40552131/99138458-d8fd2600-2645-11eb-948c-68effc08d2cf.png)
> 
> I couldn't find a custom filter to show whether a movie itself is either **"Downloaded"** or **"Not Downloaded"**.
> Do you mean the following filter?
> 
> ![image](https://user-images.githubusercontent.com/40552131/99138552-94be5580-2646-11eb-904f-ca60ffea8de4.png)

![image](https://user-images.githubusercontent.com/55031547/99138604-cd870c00-25ff-11eb-82a0-4a2e00b74861.png)

**Comment by dramirvf** [comment]
@geogolem 
![image](https://user-images.githubusercontent.com/40552131/99138595-f1217500-2646-11eb-8a9c-af57fa424201.png)

This part is the main concern. It's very useful to filter **"Missing"** items which are not currently being downloaded. For example, we may want to do a manual search for those items. Or trigger a search on **"Missing"** items which are not currently being downloaded (to avoid duplicate downloads), etc.

**Comment by geogolem** [comment]
> @geogolem
> ![image](https://user-images.githubusercontent.com/40552131/99138595-f1217500-2646-11eb-8a9c-af57fa424201.png)
> 
> This part is the main concern. It's very useful to filter **"Missing"** items which are not currently being downloaded. For example, we may want to do a manual search for those items. Or trigger a search on **"Missing"** items which are not currently being downloaded (to avoid duplicate downloads), etc.

I understand - but for me personally.. if I want to operate on stuff that is currently Downloading, I use the Activity Queue.

and stuff that is Downloading is generally only in that state for a relatively short period of time... so its usually either manageable just by inspection or 1 by 1.. or I can wait until stuff finishes downloading and carry on etc.

Having said all that I just tried to filter, Missing and it does show the Downloaded.. so I do see the issue.

I do think there should be a way to filter based on QueueState or something.. i think its called..

**Comment by dramirvf** [comment]
> > I couldn't find a custom filter to show whether a movie itself is either **"Downloaded"** or **"Not Downloaded"**.
> > Do you mean the following filter?
> 
> No - for downloaded or not use a filter like in the image below.
> 
> ![image](https://user-images.githubusercontent.com/55031547/99138604-cd870c00-25ff-11eb-82a0-4a2e00b74861.png)
> 
> The ConsideredAvailable is whether or not Radarr considers the movie available based on it's MinimumAvailability Specification and the delay. For example, if ConsideredAvailable is true, Radarr will try to download the movie during its automatic searches.
> If it is false, Radarr won't bother searching for it...

Ok. In that case the following statement does not make sense:

![Missing+Downloaded or Missing+Not Downloaded](https://user-images.githubusercontent.com/40552131/99138813-b3bde700-2648-11eb-8863-108b69a4adf0.jpg)

Because if an item is **"Missing"** it is definitely **"Not Downloaded"**. And if an item is **"Downloaded"** it is definitely **"Not Missing"**.

So what is the advantage of such filter as : **"Missing+Downloaded"** or **"Missing+Not Downloaded"** ?

**Comment by geogolem** [comment]
Lets just clarify the default filters:

All -- self explanatory
Monitored Only ( self explanatory)
Unmonitored (self explanatory)
Missing --> Monitored + Not Downloaded
Wanted --> Missing (Monitored + Not Downloaded) + ConsideredAvailable

Those are the default premade filters... 

Ya you are right about the Missing+Downloaded thing (I was just confused) , but Missing+Not Downloaded is something because it also takes into account the Monitored state.

i dont use any of these filters anyway.. I typically just try to download Movies that are "Wanted"

**Comment by dramirvf** [comment]
> > @geogolem
> > ![image](https://user-images.githubusercontent.com/40552131/99138595-f1217500-2646-11eb-8a9c-af57fa424201.png)
> > This part is the main concern. It's very useful to filter **"Missing"** items which are not currently being downloaded. For example, we may want to do a manual search for those items. Or trigger a search on **"Missing"** items which are not currently being downloaded (to avoid duplicate downloads), etc.
> 
> I understand - but for me personally.. if I want to operate on stuff that is currently Downloading, I use the Activity Queue.
> 
> and stuff that is Downloading is generally only in that state for a relatively short period of time... so its usually either manageable just by inspection or 1 by 1.. or I can wait until stuff finishes downloading and carry on etc.
> 
> Having said all that I just tried to filter, Missing and it does show the Downloading.. and i think this is right as it is.. so I do see the issue.. I understand @austinwbest comment regarding if it is Downloading it isn't really Missing anymore, but that is not how Missing works..
> 
> I do think there should be a way to filter based on QueueState (I think that is what its called internally)...

The main issue is that the **"Transient Downloading Status"** is not necessarily **"A Short Time"**. For example, I have 100+ simultaneous downloads through different download clients. Many of the torrents have few seeds and take a long time to finish. Furthermore, my download clients work only 5 hours per day. On top of that, I have a bandwidth limit of **"10 Mbps"** for download clients.

In such scenario (which is not uncommon among users), most of the items with few seeds take 2-3 week before either:
1. The download is complete
2. You come to conclusion that the torrent is fake/useless.

**Comment by geogolem** [comment]
> > > @geogolem
> > > ![image](https://user-images.githubusercontent.com/40552131/99138595-f1217500-2646-11eb-8a9c-af57fa424201.png)
> > > This part is the main concern. It's very useful to filter **"Missing"** items which are not currently being downloaded. For example, we may want to do a manual search for those items. Or trigger a search on **"Missing"** items which are not currently being downloaded (to avoid duplicate downloads), etc.
> > 
> > 
> > I understand - but for me personally.. if I want to operate on stuff that is currently Downloading, I use the Activity Queue.
> > and stuff that is Downloading is generally only in that state for a relatively short period of time... so its usually either manageable just by inspection or 1 by 1.. or I can wait until stuff finishes downloading and carry on etc.
> > Having said all that I just tried to filter, Missing and it does show the Downloading.. and i think this is right as it is.. so I do see the issue.. I understand @austinwbest comment regarding if it is Downloading it isn't really Missing anymore, but that is not how Missing works..
> > I do think there should be a way to filter based on QueueState (I think that is what its called internally)...
> 
> The main issue is that the **"Transient Downloading Status"** is not necessarily **"A Short Time"**. For example, I have 100+ simultaneous downloads through different download clients. Many of the torrents have few seeds and take a long time to finish. Furthermore, my download clients work only 5 hours per day. On top of that, I have a bandwidth limit of **"10 Mbps"** for download clients.
> 
> In such scenario (which is not uncommon among users), most of the items with few seeds take 2-3 week before either:
> 
> 1. The download is complete
> 2. You come to conclusion that the torrent is fake/useless.

Ya ya, I know . Im never really said its useless - i just said not useful for me.. I do see how it *might* be useful to some people though, which is why I have been discussing it... 

Its my bad that I thought it was a non-issue in response to austins comment.. I misunderstood.. He meant that most people don't consider it missing anymore if its downloading.. but that isn't how it actually works, i realized (kind of stupidly afterwards because it was obvious, since you were talking about the Missing filter)

So basically after all this discussion I have boiled this issue into 2 components:

1) Status column is not properly sorting

2) no ability to create custom filters based on QueueState...

I'm quickly looking through the code now to see how feasible it is to add (2).. I haven't even looked into reproducing (1).

Sorry for the confusion/ long discussion but its good that we have it to sort it out properly..

**Comment by dramirvf** [comment]
> > > > @geogolem
> > > > ![image](https://user-images.githubusercontent.com/40552131/99138595-f1217500-2646-11eb-8a9c-af57fa424201.png)
> > > > This part is the main concern. It's very useful to filter **"Missing"** items which are not currently being downloaded. For example, we may want to do a manual search for those items. Or trigger a search on **"Missing"** items which are not currently being downloaded (to avoid duplicate downloads), etc.
> > > 
> > > 
> > > I understand - but for me personally.. if I want to operate on stuff that is currently Downloading, I use the Activity Queue.
> > > and stuff that is Downloading is generally only in that state for a relatively short period of time... so its usually either manageable just by inspection or 1 by 1.. or I can wait until stuff finishes downloading and carry on etc.
> > > Having said all that I just tried to filter, Missing and it does show the Downloading.. and i think this is right as it is.. so I do see the issue.. I understand @austinwbest comment regarding if it is Downloading it isn't really Missing anymore, but that is not how Missing works..
> > > I do think there should be a way to filter based on QueueState (I think that is what its called internally)...
> > 
> > 
> > The main issue is that the **"Transient Downloading Status"** is not necessarily **"A Short Time"**. For example, I have 100+ simultaneous downloads through different download clients. Many of the torrents have few seeds and take a long time to finish. Furthermore, my download clients work only 5 hours per day. On top of that, I have a bandwidth limit of **"10 Mbps"** for download clients.
> > In such scenario (which is not uncommon among users), most of the items with few seeds take 2-3 week before either:
> > 
> > 1. The download is complete
> > 2. You come to conclusion that the torrent is fake/useless.
> 
> Ya ya, I know . Im never really said its useless - i just said not useful for me.. I do see how it _might_ be useful to some people though, which is why I have been discussing it...
> 
> Its my bad that I thought it was a non-issue in response to austins comment.. I misunderstood.. i realized (kind of stupidly afterwards because it was obvious, since you were talking about the Missing filter)
> 
> So basically after all this discussion I have boiled this issue into 2 distinct problems:
> 
> 1. Status column is not properly sorting
> 2. no ability to create custom filters based on QueueState...
> 
> I'm quickly looking through the code now to see how feasible it is to add (2) (it might be a quick change - but either way, we can do it, its just a matter of when -- im looking now) .. I haven't even looked into reproducing (1).
> 
> Sorry for the confusion/ long discussion but its good that we have it to sort it out properly..

That was exactly the point. Thanks.

**Comment by geogolem** [comment]
ya, the confusion arises because, when it is finally implemented, the custom filter will really have nothing to do with the Status column..  

We realize the *Status* column can be kind of confusing.. because there is Release Status, .. I thought about renaming the Status column to StatusSummary or something.. but I dont know.. 

There was discussion about literally just having a Status custom filter, but almost everything related to it can be filtered by other means in the custom filters, and the Status that is in that column doesn't really directly correspond to anything in radarr, it is constructed by various other properties... 

Either way, I seem to understand what you want, and possibly how to go about doing it.

**Comment by geogolem** [comment]
I started some work on this: https://github.com/geogolem/Radarr/commit/7d033c76bf416add271757a7af5799787114f206

still a WIP.

**Comment by popcorn9499** [comment]
This looks like the same issue as what I am experiencing when I am manually searching for a copy of a movie. I go to sort like I would on radarr v2 and the page pauses for a bit as if its trying to sort, however it just doesn't actually update anything on the page.

**Comment by bakerboy448** [comment]
> This looks like the same issue as what I am experiencing when I am manually searching for a copy of a movie. I go to sort like I would on radarr v2 and the page pauses for a bit as if its trying to sort, however it just doesn't actually update anything on the page.
@popcorn9499 
that is not related to the Movie Index (assuming you're on a Movie's Search tab)
Support Requests -> [Reddit](https://www.reddit.com/r/radarr/) or [Discord](https://discord.gg/r5wJPt9)
it sounds like it may be this https://wiki.servarr.com/Radarr_FAQ#Why_doesn.27t_Radarr_work_behind_an_nginx_reverse_proxy

**Comment by popcorn9499** [comment]
> > This looks like the same issue as what I am experiencing when I am manually searching for a copy of a movie. I go to sort like I would on radarr v2 and the page pauses for a bit as if its trying to sort, however it just doesn't actually update anything on the page.
> 
> @popcorn9499
> that is not related to the Movie Index (assuming you're on a Movie's Search tab)
> Support Requests -> [Reddit](https://www.reddit.com/r/radarr/) or [Discord](https://discord.gg/r5wJPt9)
> it sounds like it may be this https://wiki.servarr.com/Radarr_FAQ#Why_doesn.27t_Radarr_work_behind_an_nginx_reverse_proxy

it looked similar to me because it looked like a sorting issue due to columns not actually respecting the sorted property which you apply. However I will create a new issue if you want me to.

I am pretty much positive my configuration of radarr doesn't use nginx or any reverse proxy. I never personally did. I am just using the docker image `ghcr.io/linuxserver/radarr:latest` to run radarr from. 

Thanks for the advice however.

**Comment by austinwbest** [comment]
You don't need another issue, you where correct with it being the same.

**Comment by anzerman** [comment]
Just chiming in that I would love to be able to create custom filters based on the status field. 

At the moment there isn't really a good way to filter down to files that are a specific 'quality' (720 1080 2160). 

You can kind of use filesize, although again this is dependant on duration of the film as well.

As a work around at the moment I have added quality to the folder naming and am using the path field as a filter... Not ideal.

**Comment by SgtSeamonkey** [comment]
> Just chiming in that I would love to be able to create custom filters based on the status field.
> 
> At the moment there isn't really a good way to filter down to files that are a specific 'quality' (720 1080 2160).
> 
> You can kind of use filesize, although again this is dependant on duration of the film as well.
> 
> As a work around at the moment I have added quality to the folder naming and am using the path field as a filter... Not ideal.

This sounds like a separate feature request. You should probably make a separate issue for this. This topic is centered around a bug (Status column, internally called QueueStatus, is not sortable between "Missing", "Downloading", "Error", "Paused", etc.) and a directly related feature request (Custom filters added for QueueStatus). It's a feature that I also would like and the bug seems to still be present at the time of this post. Is this still being worked on?

**Comment by tekmyster** [comment]
i agree, this seems like a bug when sorting by status, downloading is mixed in with missing.  a custom filter would resolve the confusion
